src/lab14.py

```diff
@@ -1723,9 +1712,11 @@ def reset_zoom(self):
         active_tab.task_runner.schedule_task(task)
 
     def load(self, url):
+        self.lock.acquire(blocking=True)
```

Comment:
If this is a bug, fix the previous chapters also?

Reply:
I'm not sure if it's a bug—honestly, the locking discipline in `Browser` is a bit confusing to me. Let's discuss.

Reply:
ok...why did you add it then? You have to be careful to make sure that all callsites expect no locks.

Reply:
I, uh, think it's a bug, but I'm not sure. Basically, I looked over the code and I think we're not correctly locking when tabs are added or removed, which is a problem because in `commit` we do things based on the active tab. But I don't remember the locking discipline well and coming up with a reproducible test case for a locking issue seemed too hard. Let's discuss tomorrow and go from there—whatever the resolution is, we'll reflect that in this chapter.

Reply:
Ok. My recommendation is to remove this part of the change for now and I'll take care of investigating this code path comprehensively and fixing any locking bug.

The TL;DR of the locking discipline is: any code in Browser that touches data structures that are changed during commit needs to be enclosed in a lock.

Reply:
Done, I believe this is mergable.